alton/pause_event.py

```diff
@@ -0,0 +1,373 @@
+                log.info("Pausing pipeline '%s'.", pipeline_name)
+                self.gocd_client.pause_pipeline(pipeline_name, pause_reason)
+
+        pause_status = {
+            'event_id': event_id,
+            'paused': num_initial_pause_events == 0
+        }
+        log.info(
+            "add_pipeline_event: system '%s' with reason '%s' - status: %s",
+            pipeline_system, pause_reason, pause_status
+        )
+        return pause_status
+
+    def remove_pipeline_event(self, who_removed, pipeline_event_id):
+        """
+        Unpauses a pipeline event previously created by pausing the pipeline.
+
+        Arguments:
+            who_removed (str): HipChat username who removed the event.
+            pipeline_event_id (str): ID of the pipeline event to remove.
+
+        Returns:
+            status (dict): Dictionary with keys:
+                pipeline_system (str): Pipeline system name associated with event ID.
+                unpaused (bool): True if removing pause event caused pipeline system to be unpaused.
+                num_remaining_events (int): Number of pause events remaining for the pipeline system.
+
+        Raises:
+            PauseEventNotFound:
+                When the passed-in event ID is not found.
+            MultiplePauseEventsFound:
+                When the passed-in event ID has multiple S3 files.
+        """
+        # Capture the current date/time as a string.
+        current_time = datetime.now()
+        current_time_str = current_time.strftime(self.TIME_FORMAT)
+
+        # Read in the event with the specified event ID from the current pause events.
+        current_prefix = '{current_dir}{event_id}'.format(
+            current_dir=self.CURRENT_DIRECTORY,
+            event_id=pipeline_event_id
+        )
+        log.info('current_prefix=%s', current_prefix)
+        current_event_keys = list(self.pipeline_bucket.list(prefix=current_prefix))
+        if not len(current_event_keys):
+            raise PauseEventNotFound(pipeline_event_id)
+        elif len(current_event_keys) > 1:
+            raise MultiplePauseEventsFound(pipeline_event_id)
+        current_event_file = current_event_keys[0]
+        pause_data = yaml.safe_load(current_event_file.get_contents_as_string())
+
+        # Update the historical event.
+        event_dt = datetime.strptime(pause_data['time_paused'], self.TIME_FORMAT)
+        historical_pause_file_name = self._make_pause_event_filename(
+            pause_data['event_id'],
+            pause_data['time_paused'],
+            pause_data['pipeline_system']
+        )
+        history_pause_filepath = self._make_history_pause_filepath(event_dt, historical_pause_file_name)
+
+        # Sanity check to see if history file already exists - it should!
+        history_file = Key(self.pipeline_bucket, name=history_pause_filepath)
+        if not history_file.exists():
+            log.warning('History file "%s" does not exist - creating.', history_file.name)
```

Comment:
Why not throw an exception here?  This seems as bad as if we found multiple events in the current folder or were missing an event in the current folder.